fe/fe-core/src/main/java/com/starrocks/lake/StarMgrMetaSyncer.java

```diff
@@ -309,10 +311,18 @@ private boolean syncTableMetaInternal(Database db, OlapTable table, boolean forc
                 if (table.getState() != OlapTable.OlapTableState.NORMAL) {
                     return false; // table might be in schema change
                 }
+                if (table.getIsAutoBucketing()) {
```

Comment:
feel like this can't stop the race condition between this meta sync and the subpartition creation.

For example in the following threads condition

```
[metaSyncThread] 1. get all the physicalPartitions
[createSubPartitionThread] 2. setAutoBucketing(true)
[createSubPartitionThread] 3. create a sub partition. SUBPARTITION_X
[createSubPartitionThread] 4. setAutoBucketing(false)
[metaSyncThread] 5. check the sub partitions for the partition which the SUBPARTITION_X is not included in the `physicalPartitions`, but the tablets are the when doing `starOSAgent.listShard(groupId)`
[metaSyncThread] 6. shards for SUBPARTITION_X get deleted by mistake.
```

Reply:
step 3 create a sub partition. need db write lock and we already acquire db read lock here.
so that step 3 can not finish before we release db read lock.

Reply:
how about the subpartition is created after line 305 and before line 309?